src/webui/webapplication.cpp

```diff
@@ -318,7 +342,13 @@ void WebApplication::action_command_download()
             if ((url.size() == 40 && !url.contains(QRegExp("[^0-9A-Fa-f]")))
                 || (url.size() == 32 && !url.contains(QRegExp("[^2-7A-Za-z]"))))
                 url = "magnet:?xt=urn:btih:" + url;
-            BitTorrent::Session::instance()->addTorrent(url);
+
+            BitTorrent::AddTorrentParams params;
+            params.savePath = savepath;
+            params.label = label;
+            const QUrl qurl = QUrl::fromEncoded(url.toUtf8());
```

Comment:
Can you please elaborate a little more?  The only way to create a single commit in the Pull Request at this point would be to recreate the whole branch, that is, create it from master again, apply all the patches, a single commit, and then a push -f here.   Will that work?